packages/vue-app/template/components/nuxt-link.client.js

```diff
@@ -25,18 +25,23 @@ const observer = window.IntersectionObserver && new window.IntersectionObserver(
 export default {
   name: 'NuxtLink',
   extends: Vue.component('RouterLink'),
-  props: {
+  props: {<% if (router.prefetchLinks) { %>
     noPrefetch: {
       type: Boolean,
       default: false
-    }<% if (router.linkPrefetchedClass) { %>,
+    }<% } else { %>
+    prefetch: {
+      type: Boolean,
+      default: false
```

Comment:
Instead of adding a second prop, why not change the default value of the existing prefetch prop depending on the global prefetch status?

```suggestion
     default: <%= router.prefetchLinks ? 'true' : 'false' %>
```

Reply:
the existing prop is `no-prefetch`. If we want to keep only the new `prefetch` prop, it would introduce a breaking change 😕

Reply:
or maybe we can make the 2 props coexist when `router.prefetchLink` is set to `true`.
`no-prefetch` would have the same effect as `:prefetch="false"` to keep the compatibility.

Reply:
Ah right, sry missed that. In that case I think we should deprecate `no-prefetch` in favor of prefetch. If you set the default value of no-prefetch to undefined then you can check if its set and print a deprecation message (just make sure to do it once for all links and only in dev mode)

But opinions vary so I'd recommended to wait until @pi0 gives his approval on the deprecation :)

Reply:
I think it makes sense to be able to toggle both behaviors:

1. Having all links prefetched by default and disable single ones (think of ToS, Privacy and so on)
2. Having all links w/o prefetch and only prefetch a few of them.

Reply:
Maybe Im not understanding this correctly, but cant we achieve that with a single prop?

```js
prefetch: {
  default: true
}
...
// prefetched
<nuxt-link> 
// not prefetched
<nuxt-link :prefetch="false"> 

/* or */

prefetch: {
  default: false
}
..
// not prefetched
<nuxt-link> 
// prefetched
<nuxt-link :prefetch="true"> 
```

Reply:
I agree to have `prefetch` option (with a configurable default) and also a `no-prefetch` shortcut for more semantic usage && preventing breaking changes.

Reply:
@pi0 should the `no-prefetch` prop be deprecated ?

Reply:
I would say no as of better DX. At least not annoying users until nuxt3 if we even want to deprecate. /cc @Atinux

Reply:
One prop provides better DX in my mind. In that case if you set the prop explicitly that value will always be used, regardless of the global config.

Also if we have two props and you set them both: `<nuxt-link no-prefetch="true" prefetch="true">`, what happens then?

Reply:
Only first takes precedence. BTW I just noticed that vue3 is going to deprecate boolean props. So there are no DX benefits. (`<foo no-prop>` ~> `foo no-pro="true"`)

Reply:
So the is only annoying exciting users or not by this deprecation warn :)

Reply:
Something like that ?

```js
if (this.prefetch === this.noPrefetch) {
  console.warn("The `prefetch` and `no-prefetch` props can't be used simultaneously.")
}
```

I'm not convinced... 😕 And it would be difficult to detect with default values...
I think the deprecation warning is the best option 😉

Reply:
Nuxt 2 already comes with a default value of `true` for prefetch and vue 2 which allows boolean props. This has negative DX for _current_ users. Deprecation should have a valid reason. For nuxt 3 we can occentially just unsupport `no-` prop or start warning. But for nuxt 2 it is somehow annoying imo.

Reply:
If they think its annoying they would be able to turn it off with `features.deprecations = false` from the app features pr :stuck_out_tongue: 

Anyway, improving this behaviour in Nuxt 3 is ok with me!